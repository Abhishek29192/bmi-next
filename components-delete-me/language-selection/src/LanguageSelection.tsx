import React from "react";
import { Typography } from "@bmi-digital/components";
import { Icon } from "@bmi-digital/components";
import { Grid } from "@bmi-digital/components";
import { NavigationListButton } from "@bmi-digital/components";
import { SVGImport } from "@bmi-digital/svg-import";
import styles from "./LanguageSelection.module.scss";

export type LanguageSelectionItem = {
  code: string;
  icon?: string | SVGImport;
  label: string;
};

export type LanguageSelectionList = {
  label: string;
  menu: readonly LanguageSelectionItem[];
};

type LanguageSelectionProps = {
  introduction?: React.ReactNode;
  languages: readonly LanguageSelectionList[];
  forceMobile?: boolean;
};

const LanguageSelection = ({
  introduction,
  languages,
  forceMobile
}: LanguageSelectionProps) => (
  <div className={styles["LanguageSelection"]}>
    {introduction}
    {languages.map(({ label, menu: subMenu }, key) => (
      <div key={`language-group-${key}`}>
        <Typography className={styles["heading"]} variant="h6">
          {label}
        </Typography>
        <Grid
          container
          spacing={0}
          alignItems="center"
          className={styles["container"]}
        >
          {subMenu.map(({ label, icon, code }, key) => (
            <Grid
              item
              key={`language-${key}`}
              xs={12}
              {...(forceMobile ? {} : { md: 3, xl: 2 })}
              className={styles["item"]}
            >
              <NavigationListButton
                className={styles["link"]}
                startIcon={
                  icon &&
                  (typeof icon === "string" ? (
                    <img
                      width="20px"
                      height="16px"
                      className={styles["LanguageIcon"]}
                      src={icon}
                    />
                  ) : (
                    <Icon
                      width="20px"
                      height="16px"
                      className={styles["LanguageIcon"]}
                      source={icon}
                    />
                  ))
                }
                action={{
                  model: "htmlLink",
                  href: `/${code}`
                }}
              >
                {label}
              </NavigationListButton>
            </Grid>
          ))}
        </Grid>
      </div>
    ))}
  </div>
);

export default LanguageSelection;
