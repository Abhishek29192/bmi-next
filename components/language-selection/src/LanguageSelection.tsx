import React from "react";
import Typography from "@bmi/typography";
import Icon from "@bmi/icon";
import Grid from "@bmi/grid";
import { NavigationListButton } from "@bmi/navigation";
import styles from "./LanguageSelection.module.scss";

export type LanguageSelectionItem = {
  code: string;
  icon?: string | SVGImport;
  label: string;
};

export type LanguageSelectionList = {
  label: string;
  menu: readonly LanguageSelectionItem[];
};

type LanguageSelectionProps = {
  introduction?: React.ReactNode;
  languages: readonly LanguageSelectionList[];
};

const LanguageSelection = ({
  introduction,
  languages
}: LanguageSelectionProps) => (
  <div className={styles["LanguageSelection"]}>
    {introduction}
    {languages.map(({ label, menu: subMenu }, key) => (
      <div key={`language-group-${key}`}>
        <Typography className={styles["heading"]} variant="h6">
          {label}
        </Typography>
        <Grid container spacing={2}>
          {subMenu.map(({ label, icon, code }, key) => (
            <Grid item key={`language-${key}`} xs={12} lg={3} xl={2}>
              <NavigationListButton
                className={styles["link"]}
                startIcon={
                  icon &&
                  (typeof icon === "string" ? (
                    <img className={styles["LanguageIcon"]} src={icon} />
                  ) : (
                    <Icon className={styles["LanguageIcon"]} source={icon} />
                  ))
                }
                action={{
                  model: "htmlLink",
                  href: `/${code}`
                }}
              >
                {label}
              </NavigationListButton>
            </Grid>
          ))}
        </Grid>
      </div>
    ))}
  </div>
);

export default LanguageSelection;
