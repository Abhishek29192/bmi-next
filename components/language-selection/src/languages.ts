import GBIconSource from "./images/en_GB.svg";
import ITIconSource from "./images/it_IT.svg";

export const defaultLanguage = {
  code: "GB",
  label: "United Kingdom",
  icon: GBIconSource
};

export default [
  {
    label: "Europe",
    menu: [
      defaultLanguage,
      {
        code: "IT",
        label: "Italia",
        icon: ITIconSource
      }
    ]
  }
];
