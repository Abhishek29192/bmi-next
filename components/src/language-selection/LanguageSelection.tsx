import React from "react";
import { SVGImport } from "@bmi-digital/svg-import";
import Icon from "../icon";
import Grid from "../grid/Grid";
import Typography from "../typography/Typography";
import { NavigationListButton } from "../navigation";
import styles from "./LanguageSelection.module.scss";

export type LanguageSelectionItem = {
  code: string;
  icon?: string | SVGImport;
  label: string;
};

export type LanguageSelectionList = {
  label: string;
  menu: readonly LanguageSelectionItem[];
};

type LanguageSelectionProps = {
  introduction?: React.ReactNode;
  languages: readonly LanguageSelectionList[];
  forceMobile?: boolean;
  onCountrySelection?: (label: string, code: string) => void;
};

const LanguageSelection = ({
  introduction,
  languages,
  forceMobile,
  onCountrySelection
}: LanguageSelectionProps) => (
  <div className={styles["LanguageSelection"]}>
    {introduction}
    {languages.map(({ label, menu: subMenu }, key) => (
      <div key={`language-group-${key}`}>
        <Typography className={styles["heading"]} variant="h6">
          {label}
        </Typography>
        <Grid
          container
          spacing={0}
          alignItems="center"
          className={styles["container"]}
        >
          {subMenu.map(({ label, icon, code }, key) => (
            <Grid
              item
              key={`language-${key}`}
              xs={12}
              {...(forceMobile ? {} : { md: 3, xl: 2 })}
              className={styles["item"]}
            >
              <NavigationListButton
                data-gtm={JSON.stringify({
                  id: "nav-country-selector",
                  label,
                  action: code
                })}
                className={styles["link"]}
                startIcon={
                  icon &&
                  (typeof icon === "string" ? (
                    <img
                      width="20px"
                      height="16px"
                      className={styles["LanguageIcon"]}
                      src={icon}
                    />
                  ) : (
                    <Icon
                      width="20px"
                      height="16px"
                      className={styles["LanguageIcon"]}
                      source={icon}
                    />
                  ))
                }
                action={{
                  model: "htmlLink",
                  href: `/${code}`
                }}
                onClick={() => onCountrySelection?.(label, code)}
              >
                {label}
              </NavigationListButton>
            </Grid>
          ))}
        </Grid>
      </div>
    ))}
  </div>
);

export default LanguageSelection;
