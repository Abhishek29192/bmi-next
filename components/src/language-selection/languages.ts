import al from "./images/al.svg";
import at from "./images/at.svg";

export default [
  {
    label: "Europe",
    menu: [
      {
        code: "al",
        label: "Albania",
        icon: al
      },
      {
        code: "at",
        label: "Ã–sterreich",
        icon: at
      },
      {
        code: "ch/fr-ch",
        label: "France",
        icon: at
      }
    ]
  }
];
