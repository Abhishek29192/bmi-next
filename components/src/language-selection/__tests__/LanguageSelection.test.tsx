import { fireEvent, render } from "@testing-library/react";
import React from "react";
import languages from "../languages";
import LanguageSelection from "../LanguageSelection";

describe("LanguageSelection component", () => {
  it("renders correctly", () => {
    const { container } = render(<LanguageSelection languages={languages} />);
    expect(container).toMatchSnapshot();
  });

  it("accepts introduction text", () => {
    const { container } = render(
      <LanguageSelection
        introduction={<p>Introduction</p>}
        languages={languages}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("accepts a url to the icon", () => {
    const { container } = render(
      <LanguageSelection
        introduction={<p>Introduction</p>}
        languages={[
          {
            label: "Europe",
            menu: [
              {
                code: "al",
                label: "Albania",
                icon: "path/to/al.svg"
              },
              {
                code: "at",
                label: "Österreich",
                icon: "path/to/at.svg"
              }
            ]
          }
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correctly if forceMobile", () => {
    const { container } = render(
      <LanguageSelection
        introduction={<p>Introduction</p>}
        forceMobile
        languages={[
          {
            label: "Europe",
            menu: [
              {
                code: "al",
                label: "Albania",
                icon: "path/to/al.svg"
              },
              {
                code: "at",
                label: "Österreich",
                icon: "path/to/at.svg"
              }
            ]
          }
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("push analytics on click", () => {
    const onCountrySelection = jest.fn();
    const { getByText } = render(
      <LanguageSelection
        languages={languages}
        onCountrySelection={onCountrySelection}
      />
    );
    const countrySelectionButton = getByText("Albania");
    fireEvent.click(countrySelectionButton);
    expect(onCountrySelection).toHaveBeenCalledTimes(1);
  });
});
