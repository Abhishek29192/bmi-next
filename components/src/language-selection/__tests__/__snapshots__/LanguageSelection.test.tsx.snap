// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelection component accepts a url to the icon 1`] = `
<div>
  <div
    class="LanguageSelection"
  >
    <p>
      Introduction
    </p>
    <div>
      <h6
        class="MuiTypography-root heading Typography MuiTypography-h6"
      >
        Europe
      </h6>
      <div
        class="MuiGrid-root container MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-41 MuiButton-text buttonStyles-text-42 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Albania\\",\\"action\\":\\"/al/\\"}"
            href="/al"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-44"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-45 MuiButton-iconSizeMedium"
              >
                <img
                  class="LanguageIcon"
                  height="16px"
                  src="path/to/al.svg"
                  width="20px"
                />
              </span>
              Albania
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-41 MuiButton-text buttonStyles-text-42 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Österreich\\",\\"action\\":\\"/at/\\"}"
            href="/at"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-44"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-45 MuiButton-iconSizeMedium"
              >
                <img
                  class="LanguageIcon"
                  height="16px"
                  src="path/to/at.svg"
                  width="20px"
                />
              </span>
              Österreich
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LanguageSelection component accepts introduction text 1`] = `
<div>
  <div
    class="LanguageSelection"
  >
    <p>
      Introduction
    </p>
    <div>
      <h6
        class="MuiTypography-root heading Typography MuiTypography-h6"
      >
        Europe
      </h6>
      <div
        class="MuiGrid-root container MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-21 MuiButton-text buttonStyles-text-22 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Albania\\",\\"action\\":\\"/al/\\"}"
            href="/al"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-24"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-25 MuiButton-iconSizeMedium"
              >
                <svg
                  class="LanguageIcon"
                  height="16px"
                  width="20px"
                />
              </span>
              Albania
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-21 MuiButton-text buttonStyles-text-22 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Österreich\\",\\"action\\":\\"/at/\\"}"
            href="/at"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-24"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-25 MuiButton-iconSizeMedium"
              >
                <svg
                  class="LanguageIcon"
                  height="16px"
                  width="20px"
                />
              </span>
              Österreich
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-21 MuiButton-text buttonStyles-text-22 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"France\\",\\"action\\":\\"/ch/fr-ch/\\"}"
            href="/ch/fr-ch"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-24"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-25 MuiButton-iconSizeMedium"
              >
                <svg
                  class="LanguageIcon"
                  height="16px"
                  width="20px"
                />
              </span>
              France
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LanguageSelection component renders correctly 1`] = `
<div>
  <div
    class="LanguageSelection"
  >
    <div>
      <h6
        class="MuiTypography-root heading Typography MuiTypography-h6"
      >
        Europe
      </h6>
      <div
        class="MuiGrid-root container MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-1 MuiButton-text buttonStyles-text-2 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Albania\\",\\"action\\":\\"/al/\\"}"
            href="/al"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-4"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-5 MuiButton-iconSizeMedium"
              >
                <svg
                  class="LanguageIcon"
                  height="16px"
                  width="20px"
                />
              </span>
              Albania
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-1 MuiButton-text buttonStyles-text-2 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Österreich\\",\\"action\\":\\"/at/\\"}"
            href="/at"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-4"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-5 MuiButton-iconSizeMedium"
              >
                <svg
                  class="LanguageIcon"
                  height="16px"
                  width="20px"
                />
              </span>
              Österreich
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-xl-2"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-1 MuiButton-text buttonStyles-text-2 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"France\\",\\"action\\":\\"/ch/fr-ch/\\"}"
            href="/ch/fr-ch"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-4"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-5 MuiButton-iconSizeMedium"
              >
                <svg
                  class="LanguageIcon"
                  height="16px"
                  width="20px"
                />
              </span>
              France
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LanguageSelection component renders correctly if forceMobile 1`] = `
<div>
  <div
    class="LanguageSelection"
  >
    <p>
      Introduction
    </p>
    <div>
      <h6
        class="MuiTypography-root heading Typography MuiTypography-h6"
      >
        Europe
      </h6>
      <div
        class="MuiGrid-root container MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-61 MuiButton-text buttonStyles-text-62 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Albania\\",\\"action\\":\\"/al/\\"}"
            href="/al"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-64"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-65 MuiButton-iconSizeMedium"
              >
                <img
                  class="LanguageIcon"
                  height="16px"
                  src="path/to/al.svg"
                  width="20px"
                />
              </span>
              Albania
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root buttonStyles-root-61 MuiButton-text buttonStyles-text-62 NavigationListButton link MuiButton-textPrimary"
            data-gtm="{\\"id\\":\\"nav-country-selector\\",\\"label\\":\\"Österreich\\",\\"action\\":\\"/at/\\"}"
            href="/at"
            tabindex="0"
          >
            <span
              class="MuiButton-label buttonStyles-label-64"
            >
              <span
                class="MuiButton-startIcon buttonStyles-startIcon-65 MuiButton-iconSizeMedium"
              >
                <img
                  class="LanguageIcon"
                  height="16px"
                  src="path/to/at.svg"
                  width="20px"
                />
              </span>
              Österreich
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
